src/controller/CHIPDeviceController.cpp

```diff
@@ -203,17 +205,26 @@ CHIP_ERROR DeviceController::Shutdown()
     mSystemLayer = nullptr;
     mInetLayer   = nullptr;
 
+    chip::Platform::Delete(mExchangeMgr);
+    mExchangeMgr = nullptr;
+
+    chip::Platform::Delete(mSessionMgr);
+    mSessionMgr = nullptr;
+
+    chip::Platform::Delete(mTransportMgr);
+    mTransportMgr = nullptr;
+
     if (mStorageDelegate != nullptr)
     {
         mStorageDelegate->SetStorageDelegate(nullptr);
         mStorageDelegate = nullptr;
     }
 
-    if (mSessionManager != nullptr)
+    if (mSessionMgr != nullptr)
```

Comment:
Yes, apparently not needed.  But also, why here is `mSessionMgr->SetDelegate(nullptr);` called here, but above it wasn't?

Is that missing from above?  Is this a bug?

Reply:
OK, this seems to be resolved now in 923b110d46ea55a5655b487c3152ddb33db2938e.

The delete of `mTransportMgr` has now also been removed.  Is that correct?

Reply:
I guess this is caused by yesterday's rebase, we didn't release those managers before, it looks another PR recently landed also cover this problem.

mSessionMgr->SetDelegate(nullptr); is not needed since we deleted mSessionMgr as a whole. I would leave  mStorageDelegate->SetStorageDelegate(nullptr); there since mStorageDelegate is not allocated/released within this class.